pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -384,31 +397,40 @@ const MemoryRow = ({ memorySize, memorySizeUnit, nodeMaxMemory, recommendedMemor
 
     return (
         <>
-            <label htmlFor='memory-size' className='control-label'>
-                {_("Memory")}
-            </label>
-            <FormGroup validationState={validationStateMemory} bsClass='form-group ct-validation-wrapper' controlId='memory'>
-                <MemorySelectRow id='memory-size'
-                    value={memorySize}
-                    maxValue={nodeMaxMemory && convertToUnit(nodeMaxMemory, units.KiB, memorySizeUnit)}
-                    initialUnit={memorySizeUnit}
-                    onValueChange={e => onValueChanged('memorySize', e.target.value)}
-                    onUnitChange={value => onValueChanged('memorySizeUnit', value)} />
-                <HelpBlock id="memory-size-helpblock">
-                    {validationStateMemory === "error" && <p>{validationFailed.memory}</p>}
-                    {recommendedMemoryHelpBlock}
-                    {nodeMaxMemory && <p> {cockpit.format(
-                        _("Up to $0 $1 available on the host"),
-                        Math.floor(convertToUnit(nodeMaxMemory, units.KiB, memorySizeUnit)),
-                        memorySizeUnit,
-                    )}</p>}
-                </HelpBlock>
-            </FormGroup>
+            <Expandable id='memory-details'
```

Comment:
Honestly I don't quite buy the automatic RAM default based on a OS name. It doesn't make much difference whether you install a Debian or a Cirros machine from the OS point of view -- the critical question is what you are going to do with it. We can't anticipate that, and thus I don't believe we should hide the RAM setting by default. Could we drop the expander?

Reply:
Keep in mind, that the VM memory is configurable after installation as well, so basically we are spawning a VM that would probably meet the requirements of some/most users and the OS itself (Fedora used to fail with 1G memory for example at some point which is our current default).

I am open for discussion on this, maybe @garrett can contribute as well, but the idea was to make it as easy as possible for users to spawn a VM that would just work.
Aka someone just fills in the name and the Operating system and they have an up and running VMs (and with the patch for unattended install it would also finish the installation without interaction)

Later they can just customize it, to have more memory, more disks, NICs etc.

Reply:
Another issue that we had, is that there are so many options that the users sees, @garrett wanted that we have a minimal dialog that would just give you an up and running VM without exposing you to any options that can be somehow prefillled. (see gnomeboxes)

So, if we show these by default, the dialog doesn't get reduced in size at all :( 
And I won't be able to add new options, which is going to happen. (see unattended installation, cloud init etc)

I had proposed the idea to use https://www.patternfly.org/v4/documentation/react/components/wizard/ instead. But @garrett disapproves so I didn't really have a better alternative to start reducing the size of the dialog.

Reply:
RAM is easy to increase after creation, yes -- but e. g. disk is much harder; you need to resize the qcow, resize file systems, etc. And decreasing might not even work at all. There is some inherent complexity in creating a VM -- one does that for a particular purpose, not just because it's cool and easy to create many of them with a simple click :-) I do agree that the current dialog is rather crowded, but I don't agree that the solution is to simply hide options. The machine size is a rather basic variable here, and it will not help either way -- either users miss it and often get wrong machine types, or they expand it and then we have exactly the same dialog size problem again.

I'd rather split it into two dialogs then -- e. g. the one could define the (virtual) hardware, like name, RAM, disk; and the other the software (OS, installation source, unattended install, cloud-init).

My recommendation: Leave this PR for auto-filling the RAM/disk size based on the libvirt recommendation, and discuss/do the UI changes separately.

Reply:
I can indeed leave this PR for auto-filling the RAM/disk size and not replace the memory and storage section with expanders. But in that case when a user selects an OS they will see the relevant values to get changed, which might feel a bit weird?

Reply:
I actually think it's helpful to point out the change, so that it doesn't happen silently. It's rather surprising to set the RAM size, then change the OS, which automatically changes it.